### DISCUSSION TITLE
Custom Actions in v4

### DISCUSSION THREAD

**Discussion by atomashin** [wixtoolset/issues#7301] [Questions] [answered]
Hi, I'm trying to migrate to WiX v4 and I'm having an issue with CustomActions.CA.dll creation

In my CustomActions.csproj file I have this line referencing to v3.11 files:

```
<WixCATargetsPath Condition=" '$(WixCATargetsPath)' == '' ">$(MSBuildExtensionsPath)\Microsoft\WiX\v3.x\Wix.CA.targets</WixCATargetsPath>

And at the end of the file

<Import Project="$(WixCATargetsPath)" />
```

I've tried replacing it with this file after searching a bit online for answers `%packages%\wixtoolset.sdk\4.0.0-rc.4\build\WixToolset.Sdk.targets` but no CA.dll file was created

Is there a guide/document on how it should be done? Am I looking in the wrong place for the proper file?

**Comment by rseanhall** [comment, marked as answer]
There is a sample managed CA project at https://github.com/wixtoolset/wix4/tree/develop/src/test/burn/TestData/WixIuiBaTests/ManagedCA

**Reply by barnson** [reply]
And HeatWave has a C# custom action template.

**Reply by atomashin** [reply]
Thanks, unfortunately I couldn't find how to convert it via HeatWave so I did it manually using your example and it seems to work
